README-branching.md

```diff
@@ -0,0 +1,96 @@
+  - Is not *the most cutting edge: might not contain protocol features that scheduled for the subsequent (3.0) release.
+  - Contains changes which break backwards compatibility, or changes that depend on past compatibility-breaking changes.  “Not backwards compatible” means that a daemon running this version of mina will not connect to mainnet.
+  - Major changes to the daemon, protocol, or crypto will sometimes cause backwards-compatibility breaking changes, and of course such changes need to be done with deliberation and are not to be taken lightly.  Changes to infrastructure, auxiliary develop scripts, tests, CI, are usually not be backwards compatibility breaking, and thereby should go into compatible (unless you are doing something very special and you know what you’re doing).
+  - The difference between `develop` and `berkeley` is that `berkeley` will be the actual hardfork release, while `develop` is subsequent softfork release candidate, softfork after `berkeley`. `develop` is just not tested as rigorously, but it's softfork compatible with `berkeley`. So if `berkeley` can be thought of as 2.0, then `develop` is 2.01.
+- `o1js-main`: compatible with testnet, but has latest `proof-systems` features so that they can be used in `o1js`
+  - Contains mina changes from `rampup`
+  - But `proof-systems/develop` which by default is used by `mina/develop`.
+  - Uses `o1js/main` and `o1js-bindings/main` as explained [here](https://github.com/o1-labs/o1js/blob/main/README-dev.md#branch-compatibility?).
+  - When `proof-systems/develop` is too cutting-edge and the adaptations of its changes haven't been landed in mina, `o1js` will use the `proof-systems/o1js-main` branch which is lagging behind `proof-systems/develop` a bit.
+
+
+The relationship between the branches is as presented: `master ⊆ compatible ⊆ rampup ⊆ berkeley ⊆ develop`.
+- This means `compatible` includes all the changes in `master`, `rampup` all the changes in `compatible` and so on. So `develop` contains all the changes from *all* the stable branches, but also contains features that do not exist in any of the "subsets".
+- The back-merging direction is thus left-to-right: whenever a feature lands in this chain, it has to be periodically "back-merged" to the right.
+- The branches are merged in the other direction (upstream) only when released.
+- When merely a new feature is introduced, it should aim at the exact target branch. This place depends on the feature, e.g. `compatible` for softfork features, `develop` for more experimental/next release, etc. And then the merged feature is back-propagated downstream (to the right).
+
+
+![Illustration of the branching strategy](docs/res/branching_flow_david_wong.png)
+
+
+
+### Hard forks / releases:
+
+Whenever a hard fork happens, the code in the corresponding release branch, e.g. `berkeley`, is released.  When this happens, the current `compatible` is entirely discarded and a new `compatible` gets created based off of `develop`:
```

Comment:
Do you mean `berkeley` here?

Reply:
Suggest use bullet-list above in the doc instead of plain text as now. This is much clearer in the bullet list form.